pinot-spi/src/main/java/org/apache/pinot/spi/env/CommonsConfigurationUtils.java

```diff
@@ -287,4 +342,42 @@ private static PropertiesConfiguration createPropertiesConfiguration(boolean set
 
     return config;
   }
+
+  /**
+   * Creates the IOFactory based on the provided kind.
+   *
+   * @param ioFactoryKind IOFactory kind
+   * @return IOFactory
+   */
+  private static IOFactory createPropertyIOFactory(PropertyIOFactoryKind ioFactoryKind) {
+    switch (ioFactoryKind) {
+      case ConfigFileIOFactory:
+        return new ConfigFilePropertyIOFactory();
+      case SegmentMetadataIOFactory:
+        return new SegmentMetadataPropertyIOFactory();
+      default:
+        return new PropertiesConfiguration.DefaultIOFactory();
+    }
+  }
+
+  /**
+   * checks whether the segment metadata file first line is version header or not.
+   * @param segmentMetadataFile segment metadata file
+   * @return boolean
+   * @throws IOException exception.
+   */
+  private static boolean checkHeaderExistsInSegmentMetadata(File segmentMetadataFile) throws IOException {
+    Preconditions.checkNotNull(segmentMetadataFile,
```

Comment:
do we need to do null check? I don't feel so, but if there is need, let's annotate `@Nullable File segmentMetadataFile`.